tests/serverpod_test_server/test/relation_test.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:test/test.dart';
+
+import 'config.dart';
+
+Future<void> _createTestDatabase(Client client) async {
+  // Towns
+  var stockholm = Town(name: 'Stockholm');
+  var skinnskatteberg = Town(name: 'Skinnskatteberg');
+  stockholm.id = await client.town.insert(stockholm);
+  skinnskatteberg.id = await client.town.insert(skinnskatteberg);
+
+  // Companies
+  var serverpod = Company(name: 'Serverpod', townId: stockholm.id!);
+  var systemair = Company(name: 'Systemair', townId: skinnskatteberg.id!);
+  serverpod.id = await client.company.insert(serverpod);
+  systemair.id = await client.company.insert(systemair);
+
+  // Citizens
+  var alex = Citizen(
+      name: 'Alex', companyId: serverpod.id!, oldCompanyId: systemair.id!);
+  var isak = Citizen(name: 'Isak', companyId: serverpod.id!);
+  alex.id = await client.citizen.insert(alex);
+  isak.id = await client.citizen.insert(isak);
+}
+
+void main() async {
+  var client = Client(serverUrl);
+
+  await _createTestDatabase(client);
+
+  var citizensWithDeepIncludes = await client.citizen.getAllWithDeepIncludes();
+  var citizensShallow = await client.citizen.getAllShallow();
+
+  tearDownAll(() async {
+    await client.citizen.deleteAll();
+    await client.company.deleteAll();
+    await client.town.deleteAll();
+  });
+
+  group(
+      'Given entities with nested relations when fetching all citizens including related data ordered by id',
+      () {
+    test('then the correct number of citizens are returned.', () {
+      expect(citizensWithDeepIncludes.length, 2);
+    });
+
+    group('then first citizen fetched', () {
+      test('has correct name.', () {
```

Comment:
Suggestion: use a different wording than `correct` as this is too general, instead a better description could be `has prepopulated name.` or `has predefined name.`

Reply:
This comment applies to all tests in this file!